.github/workflows/tests.yml

```diff
@@ -0,0 +1,29 @@
+name: Run Tests
+
+on: [workflow_dispatch, push, pull_request]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.8']
```

Comment:
I think even better would be to remove the install-package action and instead just run the tests for all 3 OSes (and potentially for multiple python versions like 3.7, 3.8, 3.9, 3.10 --- or maybe just the newest 3, or just the oldest and newest supported ones).  This would obviously make sure it can install on each platform and also make sure that all code runs correctly on all platforms too.  It would make the CI pipeline take longer, but only marginally if they all run in parallel and get started at similar times.

It may require some small modifications to some tests to make sure the result being compared is agnostic to the OS being used, but that is probably best practice anyway.  If any tests are truly unique to an OS, then they can be marked with pytest to only run under certain OSes.

Reply:
Good idea!